Dristi_adhikari_Assignment 2_Final.Rmd

```diff
@@ -0,0 +1,101 @@
+---
+title: "DristiAdhikari_Assignment2-Final"
+author: "DA"
+date: "October 22, 2018"
+output: html_document
+---
+
+```{r}
+#Part1
+library(igraph)
+D1 <- read.csv("discipline-data.csv", header = TRUE)
+D1$stid <- as.factor(D1$stid)
+library(dplyr)
+D2 <- dplyr::select(D1, tid, stid)
+EDGE <- dplyr::count(D2, tid, stid)
+names(EDGE) <- c("from", "to", "count")
+V.TCH <- dplyr::select(D1, tid, t.gender, t.expertise)
+V.TCH <- unique(V.TCH)
+V.TCH$group <- "teacher"
+V.STD <- dplyr::select(D1, stid, s.gender, s.major)
+V.STD <- unique(V.STD)
+V.STD$group <- "student"
+names(V.TCH) <- c("id", "gender", "topic", "group")
+names(V.STD) <- c("id", "gender", "topic", "group")
+VERTEX <- dplyr::bind_rows(V.TCH, V.STD)
+library(igraph)
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+plot(g,layout=layout.fruchterman.reingold)
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
+
+#Part2
+#stuent - first disciplinary count for teachers, also change the titles
+StudentED <- EDGE %>% group_by(to) %>% summarize(sum(count)) 
+names(StudentED) <- c("id","count")
+#second, for teachers, also change titles
+TeachersED <- EDGE %>% group_by(from) %>% summarize(sum(count))
```

Comment:
Congratulations, you worked it out. You may have been having trouble for an unrelated reason though, it may have been because R was treating the teachers and students as different data types (teachers as character, students as numeric). This was changed once you did the `group_by`.